src/altinn-app-frontend/src/components/message/ErrorReport.tsx

```diff
@@ -1,128 +1,145 @@
+      <FullWidthWrapper onBottom={true}>
+        <Panel
+          title={getLanguageFromKey(
+            'form_filler.error_report_header',
+            language,
+          )}
+          showIcon={false}
+          variant={PanelVariant.Error}
+        >
+          <Grid
+            container={true}
+            item={true}
+            spacing={3}
+            alignItems='flex-start'
+          >
+            <Grid
+              item
+              xs={12}
             >
-              <div
-                className='a-iconText'
-                style={{ minHeight: '60px' }}
-              >
-                <div className='a-iconText-icon'>
-                  <i
-                    className='ai ai-circle-exclamation a-icon'
-                    style={{
-                      color: '#E23B53',
-                      fontSize: '4em',
-                      marginLeft: '12px',
-                    }}
-                    aria-hidden='true'
-                  />
-                </div>
-                <h2
-                  className='a-fontReg'
-                  style={{ marginBottom: '0px', marginLeft: '12px' }}
-                >
-                  <span className='a-iconText-text-large'>
-                    {getLanguageFromKey(
-                      'form_filler.error_report_header',
-                      language,
-                    )}
-                  </span>
-                </h2>
-              </div>
-            </div>
-            <div
-              className='modal-body a-modal-body'
-              style={{ paddingTop: '0px', paddingBottom: '24px' }}
-            >
-              {hasUnmappedErrors && (
-                <div>
-                  <ul style={{ listStylePosition: 'inside' }}>
-                    {unmappedErrors.map(
-                      (error: React.ReactNode, index: number) => {
-                        return <li key={index}>{error}</li>;
-                      },
-                    )}
-                  </ul>
-                </div>
-              )}
-              {!hasUnmappedErrors && (
-                // No errors to list, show a generic error message
-                <h4
-                  className='a-fontReg'
-                  style={{ marginBottom: '12px' }}
-                >
-                  <span>
-                    {getLanguageFromKey(
-                      'form_filler.error_report_description',
-                      language,
-                    )}
-                  </span>
-                </h4>
-              )}
-            </div>
-          </div>
-        </div>
-      </div>
-    </div>
+              <ul className={classes.errorList}>
+                {errorsUnmapped.map((error: React.ReactNode, index: number) => {
+                  return <li key={`unmapped-${index}`}>{error}</li>;
+                })}
+                {errorsMapped.map((error, index) => {
+                  return (
+                    <li key={`mapped-${index}`}>
```

Comment:
Using indexes in `map` is [generally not a good idea](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318), and even if it may be fine here, it leaves something that can be copied later to a place it could be problematic.

Could we use `componentId` instead? I guess that will be unique?

```suggestion
                {errorsUnmapped.map((error: React.ReactNode) => {
                  return <li key={`unmapped-${error.componentId}`}>{error}</li>;
                })}
                {errorsMapped.map((error) => {
                  return (
                    <li key={`mapped-${error.componentId}`}>
```

Reply:
Good point! I don't think we can avoid it in the first one (unmapped errors), as they (by being unmapped) don't have `componentId`. This also raises a point that I'll have to check to make sure this works with components inside repeating groups (including clicking the error to scroll the component into view).

Reply:
Tested this out, cleaned up the code and fixed an issue I found where `FormLayoutActions.updateFocus` didn't work with component IDs in repeating groups. Also reported issue #350, which I found when cleaning this up.